plugins/nlp/nlp.hpp

```diff
@@ -30,9 +35,19 @@
 #define ECL_NLP_API DECL_IMPORT
 #endif
 
-#include "hqlplugins.hpp"
-#include "eclhelper.hpp"
-#include "nlp_eng.hpp"
-#include <sstream>
+extern "C" {
+
+#ifdef ECL_NLP_EXPORTS
+ECL_NLP_API bool getECLPluginDefinition(ECLPluginDefinitionBlock *pb);
+ECL_NLP_API void setPluginContext(IPluginContext * _ctx);
+#endif
+
+}
+
+extern "C++" {
```

Comment:
extern "c++" is redundant nd confusing.

Reply:
Probably should change this line to read 

    namespace "nlp" {

Reply:
Agreed. I'm working on this PR right now to verify it fixes the current crash problem.

Reply:
I think the changes in this PR are unrelated to the crash, but will address the unwanted tracing regarding failing to load. The crash is most likely because you are assuming that strings from ECL are null terminated, when they are not.

Reply:
It was a design decision to make it so the string would not have to have a length given that texts sent to NLP analyzers are variable in length. I don't think that is the problem with the current crash. Supposedly, this fixed the crash back a couple of months. I will talk with Gordon on all this to work out the deatils.

Reply:
Your code in nlp.cpp does not pass the length it gets from ECL to the nlp-analyser. Meaning it has no way to know how long it is.

Reply:
You are right. It assumes a null terminator from that point and thus needs to be addressed.